From 2d7c4145660a7e03c01bfcfebbb2ed6bd402afbb Mon Sep 17 00:00:00 2001
From: Avery Felts <averyfelts@aol.com>
Date: Thu, 22 Oct 2026 06:48:40 -0600
Subject: [PATCH 14/55] win32u: Enable portability enumeration for Vulkan
 instances when available.

The Vulkan loader only enumerates portability drivers such as MoltenVK when
VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR and the extension are used;
without them instance creation failed with VK_ERROR_INCOMPATIBLE_DRIVER.
---
 dlls/win32u/vulkan.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/dlls/win32u/vulkan.c b/dlls/win32u/vulkan.c
index 3265d7467c8..ba568b749e1 100644
--- a/dlls/win32u/vulkan.c
+++ b/dlls/win32u/vulkan.c
@@ -448,6 +448,12 @@ static VkResult convert_instance_create_info( struct mempool *pool, VkInstanceCr
         instance->obj.extensions.has_VK_EXT_surface_maintenance1 = 1;
     if (vulkan_funcs.host_extensions.has_VK_KHR_get_physical_device_properties2)
         instance->obj.extensions.has_VK_KHR_get_physical_device_properties2 = 1;
+    /* the Vulkan loader only enumerates portability drivers (MoltenVK) when asked to */
+    if (vulkan_funcs.host_extensions.has_VK_KHR_portability_enumeration)
+    {
+        instance->obj.extensions.has_VK_KHR_portability_enumeration = 1;
+        info->flags |= VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR;
+    }
     if (use_external_memory())
         instance->obj.extensions.has_VK_KHR_external_memory_capabilities = 1;
 
-- 
2.54.0 (Apple Git-157)

